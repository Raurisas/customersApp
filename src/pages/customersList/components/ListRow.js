import React from 'react';

const ListRow = ({ item, handleRowClick, handleRemoveClick, handleEditClick }) => {
  return (
    <div className="list-row" onClick={() => handleRowClick(item.id)}>
      <div className="info">
        <span>
          {item.metadata.fullName} ({item.email})
        </span>
      </div>
      <div className="actions">
        <div onClick={event => handleEditClick(item, event)}>EDIT</div>
        <div onClick={event => handleRemoveClick(item.id, event)}>REMOVE</div>
      </div>
    </div>
  );
};

export default ListRow;
