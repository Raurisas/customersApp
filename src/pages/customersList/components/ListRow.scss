.list-row {
  border-top: 1px solid $border-color;
  border-left: 1px solid $border-color;
  border-right: 1px solid $border-color;
  cursor: pointer;
  padding: 5px 10px;
  display: flex;
  justify-content: space-between;
  .actions {
    display: flex;
    div {
      align-items: center;
      display: flex;
      height: 100%;
      font-size: 12px;
      margin-left: 10px;
      &:hover {
        opacity: 0.6;
      }
    }
  }
  .info {
    display: flex;
    word-break: break-all;
    margin: 10px 0px;
    span {
      line-height: 26px;
      margin-right: 10px;
      display: flex;
    }
  }
  &:last-child {
    border-bottom: 1px solid $border-color;

  }
  &:hover {
    background: $list-row-hover;
  }
}