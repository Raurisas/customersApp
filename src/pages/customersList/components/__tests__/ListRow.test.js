import React from 'react';
import { shallow } from 'enzyme';
import ListRow from '../ListRow';

it('should match snapshot', () => {
  const item = {
    email: 'aurimas993@gmail.com',
    metadata: {
      fullName: 'Aurimas Jackevicius'
    }
  };

  const wrapper = shallow(
    <ListRow item={item} handleRowClick={() => {}} handleRemoveClick={() => {}} handleEditClick={() => {}} />
  );
  expect(wrapper).toMatchSnapshot();
});
