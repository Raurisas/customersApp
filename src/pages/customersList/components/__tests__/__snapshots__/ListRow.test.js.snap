// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="list-row"
  onClick={[Function]}
>
  <div
    className="info"
  >
    <span>
      Aurimas Jackevicius
       (
      aurimas993@gmail.com
      )
    </span>
  </div>
  <div
    className="actions"
  >
    <div
      onClick={[Function]}
    >
      EDIT
    </div>
    <div
      onClick={[Function]}
    >
      REMOVE
    </div>
  </div>
</div>
`;
